src/NosCore.GameObject/Providers/ItemProvider/Handlers/MinilandBellHandler.cs

```diff
@@ -0,0 +1,74 @@
+﻿using System;
+using NosCore.Configuration;
+using NosCore.Core.I18N;
+using NosCore.Data.Enumerations.I18N;
+using NosCore.Data.Enumerations.Items;
+using NosCore.GameObject.ComponentEntities.Extensions;
+using NosCore.GameObject.Networking.ClientSession;
+using ChickenAPI.Packets.Enumerations;
+using ChickenAPI.Packets.ClientPackets.Inventory;
+using ChickenAPI.Packets.ServerPackets.Chats;
+using ChickenAPI.Packets.ServerPackets.UI;
+using NosCore.Data.Enumerations.Map;
+using NosCore.GameObject.Providers.InventoryService;
+using NosCore.GameObject.Providers.MapInstanceProvider;
+using NosCore.GameObject.Providers.MinilandProvider;
+
+
+namespace NosCore.GameObject.Providers.ItemProvider.Handlers
+{
+    public class MinilandBellHandler : IEventHandler<Item.Item, Tuple<InventoryItemInstance, UseItemPacket>>
+    { 
+        private readonly IMinilandProvider _minilandProvider;
+
+        public MinilandBellHandler(IMinilandProvider minilandProvider)
+        {
+            _minilandProvider = minilandProvider;
+        }
+        public bool Condition(Item.Item item) => item.Effect == ItemEffectType.Teleport && item.EffectValue == 2;
+        public void Execute(RequestData<Tuple<InventoryItemInstance, UseItemPacket>> requestData)
+        {
+            var itemInstance = requestData.Data.Item1;
+            var packet = requestData.Data.Item2;
+
+            if (requestData.ClientSession.Character.MapInstance.MapInstanceType != MapInstanceType.BaseMapInstance)
```

Comment:
Maybe the Vehicled condition and the MapInstance type condition should be in methods to avoid code duplication

Reply:
Yes could be move to a CharacterExtension